### DISCUSSION TITLE
If I could deploy contract via Remix on zksnc

### DISCUSSION THREAD

**Discussion by louialex** [zkSync-Community-Hub/zksync-developers#127] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

I deploy contract on arb works well but I deploy same code on zksync cant work with remix, anyone know the reason hello ,I try to deploy my contract via Remix after complie fine but received this error "creation of xxx errored: Returned error: {"jsonrpc":"2.0","error":"[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":3,"message":"Failed to serialize transaction: toAddressIsNull"}}}'","id":4364414066223651}", anyone could help?

### What is the specific issue or error you're encountering?

I deploy contract on arb works well but I deploy same code on zksync cant work with remix, anyone know the reason hello ,I try to deploy my contract via Remix after complie fine but received this error "creation of xxx errored: Returned error: {"jsonrpc":"2.0","error":"[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":3,"message":"Failed to serialize transaction: toAddressIsNull"}}}'","id":4364414066223651}", anyone could help?

### Can you share the error messages or logs you're receiving, if any?

I deploy contract on arb works well but I deploy same code on zksync cant work with remix, anyone know the reason hello ,I try to deploy my contract via Remix after complie fine but received this error "creation of xxx errored: Returned error: {"jsonrpc":"2.0","error":"[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":3,"message":"Failed to serialize transaction: toAddressIsNull"}}}'","id":4364414066223651}", anyone could help?

### Have you made any recent changes to the contract before encountering this issue?

I deploy contract on arb works well but I deploy same code on zksync cant work with remix, anyone know the reason hello ,I try to deploy my contract via Remix after complie fine but received this error "creation of xxx errored: Returned error: {"jsonrpc":"2.0","error":"[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":3,"message":"Failed to serialize transaction: toAddressIsNull"}}}'","id":4364414066223651}", anyone could help?

### Are there any external libraries or contracts that your contract interacts with?

I deploy contract on arb works well but I deploy same code on zksync cant work with remix, anyone know the reason hello ,I try to deploy my contract via Remix after complie fine but received this error "creation of xxx errored: Returned error: {"jsonrpc":"2.0","error":"[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":3,"message":"Failed to serialize transaction: toAddressIsNull"}}}'","id":4364414066223651}", anyone could help?

### Can you provide the relevant portions of your contract code where the issue is occurring?

I deploy contract on arb works well but I deploy same code on zksync cant work with remix, anyone know the reason hello ,I try to deploy my contract via Remix after complie fine but received this error "creation of xxx errored: Returned error: {"jsonrpc":"2.0","error":"[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":3,"message":"Failed to serialize transaction: toAddressIsNull"}}}'","id":4364414066223651}", anyone could help?

### Have you tried to isolate the problem, and if so, what were the results?

I deploy contract on arb works well but I deploy same code on zksync cant work with remix, anyone know the reason hello ,I try to deploy my contract via Remix after complie fine but received this error "creation of xxx errored: Returned error: {"jsonrpc":"2.0","error":"[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":3,"message":"Failed to serialize transaction: toAddressIsNull"}}}'","id":4364414066223651}", anyone could help?

### What steps have you already taken to try to resolve the issue?

I deploy contract on arb works well but I deploy same code on zksync cant work with remix, anyone know the reason hello ,I try to deploy my contract via Remix after complie fine but received this error "creation of xxx errored: Returned error: {"jsonrpc":"2.0","error":"[ethjs-query] while formatting outputs from RPC '{"value":{"code":-32603,"data":{"code":3,"message":"Failed to serialize transaction: toAddressIsNull"}}}'","id":4364414066223651}", anyone could help?

**Comment by louialex** [comment]
sorry, too many didnt input place,and its must, so I put it again and again

**Reply by bxpana** [reply]
@louialex we now have support on Remix with the zkSync Plugin https://docs.zksync.io/build/tooling/ide.html#remix-ide

**Comment by dutterbutter** [comment, marked as answer]
@louialex unfortunately Remix is not yet supported, however we are actively looking to support in the future. In the meantime, you can try making use of [Atlas](https://www.atlaszk.com/) instead. For an example on how to use see Hello World example [here](https://era.zksync.io/docs/dev/building-on-zksync/hello-world.html#build-and-deploy-the-greeter-contract).

**Comment by caydiverlarachel** [comment]
It seems like the error "Failed to serialize transaction: toAddressIsNull" indicates that there might be an issue with the contract's constructor parameters or deployment process. Make sure that all required constructor parameters are provided when deploying the contract and that the deployment process is executed correctly. Additionally, ensure that the contract code is compatible with zkSync and does not contain any functionalities that might not be supported. If the issue persists, further debugging of the contract code and deployment process might be necessary.

**Comment by pobertester** [comment]
Environment: Mainnet
Issue: Error deploying contract on zkSync using Remix, specifically "Failed to serialize transaction: toAddressIsNull"
Potential Cause: Serialization issue with transaction on zkSync
Recommendation: Check contract code for any issues related to transaction serialization, and ensure compatibility with zkSync
This error typically occurs when there is an issue with the serialization of the transaction, possibly related to the contract code or its compatibility with zkSync. Double-checking the contract code and ensuring compatibility with zkSync may help resolve the issue.

**Comment by dariuscandfield** [comment]
This error typically occurs when the transaction being sent has a null to address. It seems like there might be an issue with how the transaction is being constructed or serialized. Double-check your contract deployment code, especially how you're specifying the contract address. Additionally, ensure that you're using the correct network configuration and provider when deploying to zkSync via Remix.
